### PR TITLE
validator: Raises warning if interfaces exceeds 1000 in single desire state

### PR DESCRIPTION
**Pull request by SudhanAruna**
Fixing issue #1020

### CODE REVIEW

**Comment by SudhanAruna** [comment]
Fixing issue #1020

**Comment by ffmancera** [comment]
Hi, thanks I will review it soon. There is an issue in our CI so please, ignore the errors from Travis.

**Comment by SudhanAruna** [comment]
> 
> 
> Hi, thanks I will review it soon. There is an issue in our CI so please, ignore the errors from Travis.

Thanks.

File: libnmstate/validator.py [hunk 1]
```diff
@@ -75,6 +75,14 @@ class NmstateOvsLagValueError(NmstateValueError):
 
 def validate(data, validation_schema=schema.ifaces_schema):
     data = copy.deepcopy(data)
+    num_of_interfaces = len(
+        [intface for intface in data.get(schema.Interface.KEY, ())]
+    )
+    supported_interface_limit = 1000
+    if num_of_interfaces > supported_interface_limit:
+        raise NmstateValueError(
```
Comment (cathay4t) [inline libnmstate/validator.py:83]: Please use `logging.warning` instead of raise exception.

Comment (SudhanAruna) [reply libnmstate/validator.py]: @cathay4t 
Fixed. Shall I update man page also

File: libnmstate/validator.py [hunk 2]
```diff
@@ -75,6 +75,14 @@ class NmstateOvsLagValueError(NmstateValueError):
 
 def validate(data, validation_schema=schema.ifaces_schema):
     data = copy.deepcopy(data)
+    num_of_interfaces = len(
+        [intface for intface in data.get(schema.Interface.KEY, ())]
+    )
+    supported_interface_limit = 1000
```
Comment (cathay4t) [inline libnmstate/validator.py:81]: Please make this as a constant.

File: libnmstate/validator.py [hunk 3]
```diff
@@ -75,6 +75,14 @@ class NmstateOvsLagValueError(NmstateValueError):
 
 def validate(data, validation_schema=schema.ifaces_schema):
     data = copy.deepcopy(data)
+    num_of_interfaces = len(
```
Comment (cathay4t) [inline libnmstate/validator.py:78]: Please put these lines into a function like `_validate_maximum_supported_iface_count()`.

File: libnmstate/validator.py (hunk 1 above)
Comment (cathay4t) [reply libnmstate/validator.py]: That would be great.

**Comment by SudhanAruna** [comment]
@cathay4t 
Made the requested changes.

**Comment by SudhanAruna** [comment]
Thanks for the review.

**Pull request merged**